grails-app/services/org/pih/warehouse/data/ProductAssociationDataService.groovy

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.data
+
+import org.pih.warehouse.importer.ImportDataCommand
+import org.pih.warehouse.product.ProductAssociation
+import org.pih.warehouse.product.Product
+import org.springframework.validation.BeanPropertyBindingResult
+
+
+class ProductAssociationDataService {
+
+    /**
+     * Validate product association
+     */
+    Boolean validateData(ImportDataCommand command) {
+        log.info "Validate data " + command.filename
+
+        command.data.eachWithIndex { params, index ->
+            ProductAssociation productAssociationInstance = new ProductAssociation(params)
+            productAssociationInstance.product = Product.findByProductCode(params['product.productCode'])
+            productAssociationInstance.associatedProduct = Product.findByProductCode(params['associatedProduct.productCode'])
+            params['product.id'] =  productAssociationInstance.product?.id
```

Comment:
@drodzewicz is there a need to assign an id to params if you already have a `product` and an `associatedProduct` on the `productAssociationInstance`?

Reply:
the reason for that is that I have `productAssociationInstance.product` and `productAssociationInstance.associatedProduct`  inside a  
`Boolean validateData(ImportDataCommand command) { ... }` method and after validating we move on to `importData(ImportDataCommand command) { ... }` method which is responsible for creating and saving those associations.

And so for 
`ProductAssociation firstProductAssociationInstance = new ProductAssociation(params)` 
to work properly
**params** has to have properties `product.id` and `associatedProduct.id` but instead it only has  `product.productCode` and `associatedProduct.productCode`